lesson4

KochevniKXXI opened the pull request.

KochevniKXXI committed:
Четвёртый урок

DmitriyTSt commented on Lesson_4_Zemskov/app/src/main/java/hh/school/lesson_4_zemskov/GridInfoItemViewHolder.kt:
```diff
@@ -0,0 +1,25 @@
+) : RecyclerView.ViewHolder(
+    binding.root
+) {
+    fun bind(detailInfoItem: InfoItem.DetailInfoItem, position: Int) = with(binding) {
```
позиция есть внутри вьюхолдера - bindingAdapterPosition/absoluteAdapterPosition
передавать ее сюда не надо

DmitriyTSt commented on Lesson_4_Zemskov/app/src/main/java/hh/school/lesson_4_zemskov/GridInfoItemViewHolder.kt:
```diff
@@ -0,0 +1,25 @@
+        textViewHeadline.text = detailInfoItem.title
+        textViewSupportingText.text = detailInfoItem.description
+        textViewSupportingText.setTextColor(
+            if (position <= 1) binding.root.context.resources.getColor(R.color.red_500)
```
многострочные ифы по кодстайлу нашему с фигурными скобками
а вообще лучше сделать так:
```
val textColorRes = if (position <= 1) R.color.red_500 else R.color.grey_600
textViewSupportingText.setTextColor(binding.root.context.resources.getColor(textColorRes))
```

DmitriyTSt commented on Lesson_4_Zemskov/app/src/main/res/layout/item_grid.xml:
```diff
@@ -0,0 +1,42 @@
+
+        <com.google.android.material.textview.MaterialTextView
+            android:id="@+id/textViewHeadline"
+            style="@style/TextAppearance.Material3.TitleMedium"
```
если ставишь TextAppearance то логичнее его ставить в android:textAppearance

KochevniKXXI committed:
Четвёртый урок(ред.)
1. textAppeareance вместо style
2. Используется позиция ViewHolder
3. Исправлен код-стайл if
